### PR TITLE
Added support for enabling flow logs

### PR DESCRIPTION
**Pull request by farhanangullia**
- Support for enabling flow logs by specifying create_vpc_flow_logs flag
- Support for log destination (cloudwatch logs or s3)
- Dynamically creates bucket or log group if none specified
- Provides option for enriched meta data fields in flow logs

### CODE REVIEW

**Review by tonynv — COMMENTED** [review]
Please make the necessary updates to the deploy folder also. Conditionally disabling the flow logs is fine

**Review by tonynv — CHANGES_REQUESTED** [review]
Please ensure that new inputs are exposed in the deploy folder.

**Review by tonynv — APPROVED** [review]

File: modules/vpc/variables.tf [hunk 1]
```diff
@@ -207,3 +207,45 @@ variable "create_nat_gateways_private_b" {
   type        = bool
   default     = false
 }
+
+variable "create_vpc_flow_logs" {
+  description = "Controls if VPC logs should be created for the VPC"
+  type        = bool
+  default     = true
```
Comment (jaymccon) [inline modules/vpc/variables.tf:214]: Shouldn't we default this to false ?

File: variables.tf [hunk 2]
```diff
@@ -207,3 +207,45 @@ variable "create_nat_gateways_private_b" {
   type        = bool
   default     = false
 }
+
+variable "create_vpc_flow_logs" {
+  description = "Controls if VPC logs should be created for the VPC"
+  type        = bool
+  default     = true
```
Comment (jaymccon) [inline variables.tf:214]: Shouldn't we default this to false ?

File: modules/vpc/variables.tf (hunk 1 above)
Comment (farhanangullia) [reply modules/vpc/variables.tf:214]: @tonynv Should we default it to false and leave it to the users to decide if they want to enable it?

Comment (tonynv) [reply modules/vpc/variables.tf:214]: Yes

File: variables.tf (hunk 2 above)
Comment (tonynv) [reply variables.tf:214]: Yes

File: modules/vpc/variables.tf (hunk 1 above)
Comment (farhanangullia) [reply modules/vpc/variables.tf:214]: Updated #18

File: variables.tf (hunk 2 above)
Comment (farhanangullia) [reply variables.tf:214]: Updated #18

**Pull request merged**